src/backup.ts

```diff
@@ -0,0 +1,387 @@
+   * const spanner = new Spanner();
+   * const instance = spanner.instance('my-instance');
+   * const backup = instance.backup('my-backup');
+   * const expireTime = await backup.getExpireTime();
+   * console.log(`Backup ${backup.formattedName_} expires on ${expireTime.toISOString()}`);
+   */
+  async getExpireTime(): Promise<PreciseDate | undefined> {
+    const [backupInfo] = await this.getMetadata();
+    const expireTime = backupInfo.expireTime;
+    return expireTime ? new PreciseDate(expireTime as DateStruct) : undefined;
+  }
+
+  /**
+   * Checks whether the backup exists.
+   *
+   * @see {@link #getMetadata}
+   *
+   * @method Backup#exists
+   * @returns {Promise<boolean>} when resolved, contains true if the backup
+   *     exists and false if it does not exist.
+   *
+   * @example
+   * const {Spanner} = require('@google-cloud/spanner');
+   * const spanner = new Spanner();
+   * const instance = spanner.instance('my-instance');
+   * const database = spanner.database('my-database');
+   * const backup = instance.backup('my-backup');
+   * const alreadyExists = await backup.exists();
+   * console.log(`Does backup exist? ${alreadyExists}`);
+   */
+  async exists(): Promise<boolean> {
+    try {
+      // Attempt to read metadata to determine whether backup exists
+      await this.getMetadata();
+      // Found therefore it exists
+      return true;
+    } catch (err) {
+      if (err.code === status.NOT_FOUND) {
+        return false;
+      }
+      // Some other error occurred, rethrow
+      throw err;
+    }
+  }
+
+  updateExpireTime(expireTime: PreciseDate): Promise<Backup>;
+  updateExpireTime(
+    expireTime: PreciseDate,
+    callback: UpdateExpireTimeCallback
+  ): void;
+  /**
+   * Sets the expiry time of a backup.
+   *
+   * @see {@link #getExpireTime}
+   *
+   * @method Backup#updateExpireTime
+   * @returns {Promise<Backup>} when resolved, the backup's expire time will
+   *     have been updated.
+   *
+   * @example
+   * const {Spanner} = require('@google-cloud/spanner');
+   * const spanner = new Spanner();
+   * const instance = spanner.instance('my-instance');
+   * const myBackup = instance.backup('my-backup');
+   * const oneDay = 1000 * 60 * 60 * 24;
+   * const newExpireTime = new PreciseDate(Date.now() + oneDay);
+   * await myBackup.updateExpireTime(newExpireTime);
+   */
+  updateExpireTime(
+    expireTime: PreciseDate,
+    callback?: UpdateExpireTimeCallback
+  ): void | Promise<Backup> {
+    this.expireTime = expireTime;
+
+    const reqOpts: databaseAdmin.spanner.admin.database.v1.IUpdateBackupRequest = {
+      backup: {
+        name: this.formattedName_,
+        expireTime: expireTime.toStruct(),
+      },
+      updateMask: {
+        paths: ['expire_time'],
+      },
+    };
+    return this.request<databaseAdmin.spanner.admin.database.v1.IBackup>(
+      {
+        client: 'DatabaseAdminClient',
+        method: 'updateBackup',
+        reqOpts,
+      },
+      err => {
```

Comment:
Is there a `response` from this request that we could just return both to the user?

Reply:
The RPC returns an `IBackup` which contains the updated expire time. Instead of returning that, @prunge-helix has chosen to return the `Backup`. I'm not sure if there's value to returning both.

I did realize one thing though. We should only update `this.expireTime` if the update was successful so I've fixed that now.

Reply:
Returning `err, response` is the usual pattern. We should keep to returning the raw response, as that could provide value to the user. Additionally, should the API response contain more information in the future, we wouldn't need to introduce a breaking change to return it instead of `this`.

Reply:
That makes sense. I updated it to return the `IBackup`.